WangscapeGui/optionseditor.h

```diff
@@ -0,0 +1,33 @@
+#ifndef OPTIONSEDITOR_H
+#define OPTIONSEDITOR_H
+
+#include <QDockWidget>
+
+#include <Options.h>
+
+namespace Ui {
+class OptionsEditor;
+}
+
+class OptionsEditor : public QDockWidget
+{
+    Q_OBJECT
+
+public:
+    explicit OptionsEditor(QWidget *parent_ = 0);
+    ~OptionsEditor();
+    bool useDefaultOuputDir() const;
+    void setOptions(Options* options);
+
+private slots:
+    void setAlphaCalculatorMode(const QString &mode_name);
+    void switchDefaultOutputDirFlag(int newCheckboxState);
+
+private:
+    void setAlphaCalculatorModeComboBox();
+    Ui::OptionsEditor *mUi;
+    Options* mOptions;
```

Comment:
Smart pointers please.

Reply:
Yes, you're right here. I thought that it's not owning pointer and this class is managed by it's parent, `MainWindow`, so it will be ok, but:
* it may change in the future and
* storing it in `std::shared_ptr` in this place is just a bit safer :smiley:.